app/helpers/ads_helper.rb

```diff
@@ -30,6 +30,10 @@ def kinds_for_select
     Ad.kinds.keys.map { |key| [t("ad.kinds.#{key}"), key] }
   end
 
+  def cities_for_filter_select
+    City.where(id: Ad.select("DISTINCT(city_id)")).order(:name).pluck(:name, :id)
```

Comment:
I played around a bit and found that this query does the same thing, but in one db query as opposed to two:
```rb
City.joins(:ads).all.pluck(:name, :id).uniq
```

For this to work, you need to add `has_many :ads` to City model.

Reply:
`City.where(id: Ad.select("DISTINCT(city_id)")).order(:name)` also generates on query (with subquery) and does not return duplicates (no need for uniq). Haven't tested which one is faster :)